src/hazelcore/semantics/Action.re

```diff
@@ -3154,19 +3242,29 @@ module Exp = {
 
     | (Delete, CursorE(OnText(j), Var(_, _, x))) =>
       syn_delete_text(ctx, u_gen, j, x)
-    | (Delete, CursorE(OnText(j), NumLit(_, n))) =>
-      syn_delete_text(ctx, u_gen, j, string_of_int(n))
+    | (Delete, CursorE(OnText(j), IntLit(_, n))) =>
+      syn_delete_text(ctx, u_gen, j, n)
+    | (Delete, CursorE(OnText(j), FloatLit(_, f))) =>
+      syn_delete_text(ctx, u_gen, j, f)
     | (Delete, CursorE(OnText(j), BoolLit(_, b))) =>
       syn_delete_text(ctx, u_gen, j, string_of_bool(b))
-
     | (Backspace, CursorE(OnText(j), Var(_, _, x))) =>
       syn_backspace_text(ctx, u_gen, j, x)
-    | (Backspace, CursorE(OnText(j), NumLit(_, n))) =>
-      syn_backspace_text(ctx, u_gen, j, string_of_int(n))
+    | (Backspace, CursorE(OnText(j), IntLit(_, n))) =>
+      syn_backspace_text(ctx, u_gen, j, n)
+    /* For Float: (.x<|) ==> ( |_ )  */
+    | (Backspace, CursorE(OnText(2), FloatLit(_, f)))
+        when
+          FloatUtil.less_than_one_str(f) && FloatUtil.num_digits_str(f) == 2 =>
```

Comment:
This needs to be a little more general. For example, right now we don't handle backspacing on `1|.` or deleting on `.|1`.

Reply:
means delete the whole thing when there is only a dot left